src/test/scala/fable/ConsumerSpec.scala

```diff
@@ -10,30 +10,32 @@ import org.apache.kafka.clients.producer.{
   ProducerConfig,
   ProducerRecord
 }
+import org.apache.kafka.common.TopicPartition
 import org.apache.kafka.common.serialization.StringSerializer
 import org.scalatest.AsyncFunSuite
+import org.apache.kafka.clients.consumer.{MockConsumer, OffsetResetStrategy}
 import scala.concurrent.ExecutionContext
+import scala.collection.mutable.ArrayBuffer
+import scala.collection.JavaConverters._
 
 class ConsumerSpec extends AsyncFunSuite {
   implicit val contextShift = IO.contextShift(implicitly[ExecutionContext])
 
   test("poll") {
-    val topic = Topic("fable-test-example")
-    val consumer = Consumer.resource[IO, String, String](config)
+    val (mockConsumer, topic, topicPartition, offset) = createMockKafkaObjects
```

Comment:
What do you think about moving most of these into local values and passing them in? Making the mock method own them means you can't change them per test, and it means you lose a lot of the context out of view.